src/aave-v3/SupplyVaultBase.sol

```diff
@@ -28,52 +28,59 @@ abstract contract SupplyVaultBase is ERC4626UpgradeableSafe {
 
     /// STORAGE ///
 
-    IMorpho public morpho; // The main Morpho contract.
+    IMorpho public immutable morpho; // The main Morpho contract.
+    IPool public immutable pool; // The AaveV3 pool.
+
     address public poolToken; // The pool token corresponding to the market to supply to through this vault.
-    IPool public pool;
 
     /// @dev This empty reserved space is put in place to allow future versions to add new
     /// variables without shifting down storage in the inheritance chain.
     /// See https://docs.openzeppelin.com/contracts/4.x/upgradeable#storage_gaps
-    uint256[47] private __gap;
+    uint256[49] private __gap;
+
+    /// CONSTRUCTOR ///
+
+    /// @dev Initializes network-wide immutables
+    /// @param _morpho The address of the main Morpho contract.
+    constructor(address _morpho) {
```

Comment:
Not really related to this PR specifically so it's not something we missed, but maybe we should add events for morpho and pool being set.

Reply:
Yes, especially because they are immutables because it makes explicit where these vars were set without going into the calldata imo